incubator/hnc/pkg/reconcilers/hierarchy_config.go

```diff
@@ -99,8 +100,8 @@ func (r *HierarchyConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, er
 }
 
 func (r *HierarchyConfigReconciler) reconcile(ctx context.Context, log logr.Logger, nm string) error {
-	// Get the singleton and namespace.
-	inst, nsInst, err := r.getInstances(ctx, log, nm)
+	// Get the singleton, the namespace and a list of the names of the hns instances in the namespace.
+	inst, nsInst, hnsnms, err := r.getInstances(ctx, log, nm)
```

Comment:
If the HNS reconciler updates the in-memory forest, there shouldn't be any need to read it here.